### ISSUE TITLE
Sitemap lastmod uses last publish date instead of last page edit date

### ISSUE THREAD

**Issue by johnsicili** [webstudio-is/webstudio#3430]
Go to a sitemap (e.g. [Webstudio's](https://webstudio.is/sitemap.xml) and see that the lastmod dates are all the same value.

These need to reflect the last time the page's value(s) were updated. Currently it's taking the last publish time and using that value.

> Google uses the <lastmod> value if it's consistently and verifiably (for example by comparing to the last modification of the page) accurate. [src](https://developers.google.com/search/docs/crawling-indexing/sitemaps/build-sitemap)

**Comment by istarkov** [comment]
Technically even if content of the page is not changed but scripts are, css is changed for the whole site too. We are not cms so almost every publish has effect on the whole app. Its even without slots etc. Internally we have nothing now to detect the latest change of the page too, I would postpone this after collaboration data structures where at least we will have some kind of time for every change

**Comment by kof** [comment]
@istarkov true, lets see how webflow treats this

**Comment by kof** [comment]
We need to learn if this date should be updated when anything changes or when only content changes and then think about all kinds of content possibly changing: text, variables,  content that is fetched remotely. It is a challenging thing to solve indeed.

**Comment by kof** [comment]
@johnsicili is the date correct in case of dynamic pages where content comes from CMS?

**Comment by istarkov** [comment]
Based on my understanding, the primary reason for considering this field is to manage the "crawl budget" effectively. There is a theory that Google imposes certain constraints on how often it will recrawl your site. For websites with 100-500 pages, this is typically a small number, and worrying about the crawl budget may not be necessary. However, for larger sites, especially those without a Content Management System (CMS), managing this can be challenging. But above is not about cms at all.

**Comment by kof** [comment]
True, on a static site crowl budget isn't gonna be a problem on a dynamic cms-based one, date will be correct based on content changes.

**Comment by kof** [comment]
Thinking about it if CMS based sites report correct dates, I think we should just close this issue and not bother at all.

**Comment by johnsicili** [comment]
CMS properly reports `lastmod` because they are bound to the external value. I'm pinging an SEO group for further clarification on the subject.

**Comment by johnsicili** [comment]
BTW, Webflow does not use `lastmod`. 

https://discourse.webflow.com/t/add-and-automatically-update-lastmod-tag-in-sitemap/225322

**Comment by johnsicili** [comment]
I asked John Mueller of Google the following question:

TL;DR – If Google determines lastmod isn’t “consistently and verifiably accurate”, do they treat all URLs in the sitemap this way? Or is it conditional? If I have a batch of URLs that have accurate lastmod dates and a batch that have not so accurate, is it best to use two separate sitemaps?

He responded:

I wouldn't see the lastmod date as a magic bullet. Use it where you have something useful to share, skip it otherwise. If you do that in separate sitemap files, that might make it easier for you to maintain. It's also fine to share whatever level of accuracy that you have - it doesn't need to be millisecond-level (it can be by day).

Depending on the site, you might also consider RSS feeds, which have similar recommendations in terms of dates: https://developers.google.com/search/blog/2014/10/best-practices-for-xml-sitemaps-rssatom

---

It doesn't directly answer the question if the entire sitemap's lastmod will be disregarded if it finds a handful of them not accurate.

IMO keeping them at least has upside potential – Google will use them. If there are too many inconsistencies, then they might not use them which likely is the same as not having the date altogether. I would go for two separate sitemaps just in case though.

Another note is that lastmod should only be used if there are meaningful changes.

**Comment by kof** [comment]
Sounds like we can close the issue?
